src/server/route/routes/edge.ts

```diff
@@ -32,12 +32,20 @@ router
         console.log('Receiving edge...', req.body);
         const text: IEdge = req.body; //Might have to parse this
         try {
-            const q = await db.query(
+            const q1 = await db.query(
+                'SELECT * FROM edge WHERE source_id = $1 AND target_id = $2',
+                [text.target_id, text.source_id]
+            );
+            if (q1.rowCount > 0) {
+                throw new Error('Both-way edges are not allowed');
+            }
+
+            const q2 = await db.query(
                 'INSERT INTO edge (source_id, target_id) VALUES ($1, $2)',
```

Comment:
Näin jälkikäteen ajatellen olisi varmaan käyttäjäystävällisempää tässä virheen sijaan korvata olemassaoleva edge uudella. Tällöin käyttäjä pystyisi helposti korjaamaan dependenssien suuntia graafeissa, eikä välissä tarvitsisi poistaa edgeä.